presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -849,9 +848,13 @@ public synchronized void revokeRoles(Set<String> roles, Set<HivePrincipal> grant
 
     private String getTableOwner(HiveIdentity identity, String databaseName, String tableName)
     {
-        Table table = delegate.getTable(identity, databaseName, tableName)
+        return getSourceTableOrElseThrow(identity, databaseName, tableName).getOwner();
+    }
+
+    private Table getSourceTableOrElseThrow(HiveIdentity identity, String databaseName, String tableName)
```

Comment:
It's obvious that a method throws when it cannot return a value, so `OrElseThrow` doesn't add value.

```suggestion
    private Table getTable(HiveIdentity identity, String databaseName, String tableName)
```

Reply:
😞

Reply:
`getTable` doesn't work as it's conflicting with other `getTable`. Maybe `getSourceTable` is better: https://github.com/prestosql/presto/pull/1880